### ISSUE TITLE
Chat Widget Issue - CORB

### ISSUE THREAD

**Issue by luxD3Br1** [Mintplex-Labs/anything-llm#2561]
Context:
1. Hosted on  local docker, tunnelled using Ngrok for the website(vercel.app) chat widget. (works locally on my pc, ngrok domain can be access properly)
2. Have been receiving CORS error so i setup CORS to allow the Vercel Domain, still error but now CORB.
3. I do not know what else could be causing this aside from the content header types.

> in Postman
`http://ngrok(or)localhost:3001/embed/anythingllm-chat-widget.min.js`

This will show the content-type as `text/html` instead of I THINK `text/javascript` or something. i really do not know anymore

**Comment by timothycarambat** [comment]
You are correct that `application/javascript` should be the `Content-Type` header. The MIME's wont match and Chrome may fail to render if the type is anything else. How did you set up the vercel domain because it is surely related to that.

**Comment by timothycarambat** [comment]
If you copy the `min.js` file directly from AnythingLLM and put it local to the site to it is no longer cross-origin - do you still have this issue? The js file is served from AnythingLLM by default since that is easier, but you could take the raw file and host it anywhere

**Comment by luxD3Br1** [comment]
I know its closed but i wanna give an Update.

Changed tunnelling service to serveo instead of ngrok, it worked.
After all this failed reverse proxy and custom headers with ngrok -_-

**Comment by timothycarambat** [comment]
??? I wonder why that worked. I know NGROK has some options for [`--request-header`](https://ngrok.com/docs/http/request-headers/) - maybe more config on that would have been a resolution?

**Comment by luxD3Br1** [comment]
Ngrok imposes a browser warning page when new connections attempt to access the upstream server. This page requires users to confirm they understand they are connecting to an Ngrok domain, resulting in unexpected behavior when using Ngrok to host a chatbot widget. Specifically, each time the Vercel-hosted website requests JavaScript or CSS files required for the widget, Ngrok returns the warning page instead of the expected resources. This discrepancy causes Cross-Origin Read Blocking (CORB) issues, as the returned headers do not align with the anticipated ones.

To address this, the "ngrok-skip-browser-warning" custom header was added in several configurations: first in the Ngrok web interface's edge settings, and later in Vercel’s fetch requests to manage CORS for the Ngrok domain. The header appeared correctly in Postman tests; however, the warning page continued to appear inconsistently. Additionally, attempts were made to modify the server file source code to include this custom header and rebuild the Docker image to bypass the warning, but this also proved ineffective. [Info](https://stackoverflow.com/questions/73017353/how-to-bypass-ngrok-browser-warning#:~:text=PM%20from%20ngrok,at%207%3A43)

As a temporary solution, Servio was used, though a more stable configuration is planned for hosting the AnythingLLM instance. Efforts will also continue to refine Ngrok for static domain requests.
